mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -29,85 +93,125 @@ def path(self):
+        else:
+            self.presenter.update_image(image_files)
 
 
 class ImageWatcher(QObject):
     """
     A class to watch a directory for new images.
+
+    ...
+
+    Attributes
+    ----------
+    directory : str
+        path to directory to watch
+    watcher : QFileSystemWatcher
+        file system watcher to watch directory
+    images : list
+        list of images in directory
+    image_changed : pyqtSignal
+        signal emitted when an image is added or removed
+
+    Methods
+    -------
+    find_images()
+        Find all the images in the directory and emit the
+        image_changed signal for the last modified image.
+    sort_images_by_modified_time(images)
+        Sort the images by modified time.
+    find_last_modified_image()
+        Find the last modified image in the directory and
+        emit the image_changed signal.
     """
-    image_changed = pyqtSignal(str)  # Signal emitted when an image is added or removed
+    image_changed = pyqtSignal(list)  # Signal emitted when an image is added or removed
 
     def __init__(self, directory):
+        """
+        Constructor for ImageWatcher class which inherits from QObject.
+
+        Parameters
+        ----------
+        directory : str
+            path to directory to watch
+        """
+
         super().__init__()
         self.directory = directory
         self.watcher = QFileSystemWatcher()
         self.watcher.directoryChanged.connect(self._handle_directory_change)
         self.watcher.addPath(str(self.directory))
         self.images = []
 
-    def find_images(self):
+    def find_images(self) -> None:
         """
         Find all the images in the directory and emit the
         image_changed signal for the last modified image.
         """
         self.images = self._get_image_files()
 
-    def find_last_modified_image(self):
+    def sort_images_by_modified_time(self, images: list[Image_Data]) -> list[Image_Data]:
         """
-        Find the last modified image in the directory and
-        emit the image_changed signal.
+        Sort the images by modified time.
+
+        :param images: list of image objects to sort by modified time
+        :return: sorted list of images
         """
+        return sorted(images, key=lambda x: x.image_modified_time)
 
+    def get_images(self):
+        """Return the sorted images"""
         if self.images:
             try:
-                last_modified_image = max(self.images, key=lambda x: x.stat().st_mtime)
+                return self.images
             except FileNotFoundError:
-                last_modified_image = ''
-
-            self.image_changed.emit(str(last_modified_image))
-
-    def _handle_directory_change(self, directory):
-        self.find_images()
-        if self.images and self._validate_images(self.images):
-            self.find_last_modified_image()
-        else:
-            self.image_changed.emit('')
+                return []
```

Comment:
I don't think this can fail. `self.images` is just a list that might be empty. Think we can either just return the list, or let the caller use the attribute directly.

Reply:
The diff has shown in a strange way. The comment is for the lines
```python
    def get_images(self):
        """Return the sorted images"""
        if self.images:
            try:
                return self.images
            except FileNotFoundError:
                return []
```

Reply:
I'll keep it in it's own method for the time being in case we choose to modify to include additional logic later on.